core/field_checkbox.js

```diff
@@ -108,19 +99,30 @@ Blockly.FieldCheckbox.prototype.configure_ = function(config) {
  * @package
  */
 Blockly.FieldCheckbox.prototype.initView = function() {
-  this.size_.width = this.constants_.FIELD_CHECKBOX_DEFAULT_WIDTH;
   Blockly.FieldCheckbox.superClass_.initView.call(this);
 
-  this.textElement_.setAttribute('x', this.constants_.FIELD_CHECKBOX_X_OFFSET);
-  this.textElement_.setAttribute('y', this.constants_.FIELD_CHECKBOX_Y_OFFSET);
-  this.textElement_.removeAttribute('dominant-baseline');
-  Blockly.utils.dom.addClass(this.textElement_, 'blocklyCheckbox');
-
-  this.textContent_.nodeValue =
-      this.checkChar_ || Blockly.FieldCheckbox.CHECK_CHAR;
+  Blockly.utils.dom.addClass(
+      /** @type {!SVGTextElement} **/ (this.textElement_), 'blocklyCheckbox');
   this.textElement_.style.display = this.value_ ? 'block' : 'none';
 };
 
+/**
+ * @override
+ */
+Blockly.FieldCheckbox.prototype.render_ = function() {
+  if (this.textContent_) {
+    this.textContent_.nodeValue = this.getDisplayText_();
+  }
+  this.updateSize_(this.constants_.FIELD_CHECKBOX_X_OFFSET);
```

Comment:
Why call the parent updateSize?  Checkboxes barely need anything from it.

Reply:
Same as previous comment, checkboxes have a checkmark which is a ``blocklyText`` element and that needs updating if the font size changes.